src/ListView.js

```diff
@@ -0,0 +1,265 @@
+import DeleteCompletedPopup from "./DeleteCompletedPopup";
+import ListContents from "./ListContents";
+import Backdrop from "./Backdrop";
+import { useState, useEffect, useRef } from "react";
+import { generateUniqueID } from "web-vitals/dist/modules/lib/generateUniqueID";
+import {
+  useCollectionData,
+  useDocumentData,
+} from "react-firebase-hooks/firestore";
+import {
+  query,
+  setDoc,
+  updateDoc,
+  deleteDoc,
+  doc,
+  orderBy,
+  where,
+  serverTimestamp,
+  collection,
+} from "firebase/firestore";
+import TOP_LEVEL_COLLECTION from "./firestore-config.js";
+
+function ListView(props) {
+  const collectionRef = collection(
+    props.db,
+    TOP_LEVEL_COLLECTION,
+    props.currentList,
+    "items"
+  );
+
+  //   const [sortType, setSortType] = useState("created");
+  const [toScroll, setToScroll] = useState(false);
+
+  const [showCompleted, setShowCompleted] = useState(true);
+
+  const [deleteCompletedPopup, setDeleteCompletedPopup] = useState(false);
+  function handleDeleteCompletedPopup() {
+    setDeleteCompletedPopup(!deleteCompletedPopup);
+  }
+
+  // Use for deleting all completed items
+  const isCheckedQuery = query(collectionRef, where("checked", "==", true));
+  const [checkedData, checkedLoading, checkedError] =
+    useCollectionData(isCheckedQuery);
+
+  const metadataRef = collection(props.db, TOP_LEVEL_COLLECTION);
+  const [metadata, metadataLoading, metadataError] = useDocumentData(
+    doc(metadataRef, props.currentList)
+  );
+
+  const bottomBar = useRef();
+  function getBottomBarLocation() {
+    const rect = bottomBar.current.getBoundingClientRect();
+    return rect.top;
+  }
+
+  const [filter, setFilter] = useState("");
+
+  if (metadataError) {
+    console.log(metadataError);
+    // TODO: actual error message?
+  }
+
+  let sortType = "created";
+  let title = "Loading...";
+  // Get data from database.
+  if (!metadataLoading) {
+    sortType = metadata.sort;
+    title = metadata.text;
+  }
+
+  let orderByParam = orderBy(sortType);
+  if (sortType === "priority") {
+    orderByParam = orderBy("priority", "desc");
+  }
+  let queryParam = query(collectionRef, orderByParam);
+  if (!showCompleted) {
+    queryParam = query(
+      collectionRef,
+      orderByParam,
+      where("checked", "==", false)
+    );
+  }
+  const [data, loading, error] = useCollectionData(queryParam);
+
+  let filteredData = data;
+  if (!loading) {
+    filteredData = data.filter((item) => item.text.toLowerCase().includes(filter.toLowerCase()));
+  }
+
+  if (error) {
+    console.log(error);
+  }
+
+  function handleDeleteCompletedTasks() {
+    let completedTasks = [];
+    // TODO: ask about whether or not we should have this thru database or not
+    if (!checkedLoading && !checkedError) {
+      completedTasks = checkedData;
+    } else {
+      completedTasks = data.filter((task) => task.checked === true);
+    }
+    let deleteCounter = 0;
+    completedTasks.forEach((task) => {
```

Comment:
You have concurrency issues (with a separate instance of the app).
Imagine list with 3 tasks: a, b, c
On one machine, user completes a and b and deletes them. At same time, on second machine, user complete a and deletes it. Afterwards, total will be 0 (instead of 1) and complete will be -1.

Solution is to wrap your deleteDocs and your updateDoc all in a single Firestore transaction.
Both